crates/hir/src/lib.rs

```diff
@@ -2951,6 +3123,40 @@ impl GenericDef {
             })
             .collect()
     }
+
+    pub fn type_params(self, db: &dyn HirDatabase) -> Vec<TypeParam> {
```

Comment:
these APIs are also not so great since they lose the order of type/const params. IMO it would be better to have the caller use `type_or_const_params` and filter themselves if they're really only interested in one

Reply:
Good point, I'll do that